pkgs/desktops/budgie/budgie-desktop/default.nix

```diff
@@ -0,0 +1,99 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, accountsservice
+, alsa-lib
+, budgie-screensaver
+, docbook-xsl-nons
+, glib
+, gnome
+, gnome-desktop
+, gnome-menus
+, graphene
+, gst_all_1
+, gtk-doc
+, gtk3
+, ibus
+, intltool
+, libcanberra-gtk3
+, libgee
+, libGL
+, libnotify
+, libpeas
+, libpulseaudio
+, libuuid
+, libwnck
+, mesa
+, meson
+, ninja
+, pkg-config
+, polkit
+, sassc
+, upower
+, vala
+, wrapGAppsHook
+}:
+
+stdenv.mkDerivation rec {
+  pname = "budgie-desktop";
+  version = "10.7";
+
+  src = fetchFromGitHub {
+    owner = "BuddiesOfBudgie";
+    repo = pname;
+    rev = "v${version}";
+    fetchSubmodules = true;
+    sha256 = "sha256-ZPd9KwEFYmslOc9+B/DXdhqptgpyCC8tiag/60ReocA=";
+  };
+
+  nativeBuildInputs = [
+    docbook-xsl-nons
+    gtk-doc
+    intltool
+    meson
+    ninja
+    pkg-config
+    vala
+    wrapGAppsHook
+  ];
+
+  buildInputs = [
+    accountsservice
+    alsa-lib
+    budgie-screensaver
+    glib
+    gnome-desktop
+    gnome-menus
+    gnome.gnome-bluetooth_1_0
+    gnome.gnome-settings-daemon
+    gnome.mutter
+    graphene
+    gst_all_1.gstreamer
+    gtk3
+    ibus
+    libcanberra-gtk3
+    libgee
+    libGL
+    libnotify
+    libpeas
+    libpulseaudio
+    libuuid
+    libwnck
+    mesa
+    polkit
+    sassc
+    upower
+  ];
+
```

Comment:
Check the `Exec`/`TryExec` lines in [`org.buddiesofbudgie.BudgieDesktopNmApplet.desktop`](https://github.com/BuddiesOfBudgie/budgie-desktop/blob/main/src/session/budgie-desktop-nm-applet.desktop.in) and patch if necessary.